src/dune_rules/workspace.ml

```diff
@@ -260,17 +260,31 @@ module Context = struct
   end
 
   module Default = struct
-    type t = Common.t
+    type t =
+      { base : Common.t
+      ; lock : Path.Source.t option
```

Comment:
Would it be simpler to drop the `option` and default this field to `Dune_pkg.Lock_dir.path` during parsing if no value is specified in the workspace? Is it ever desirable for dune to distinguish between the case where `lock` is omitted and the case where it's explicitly set to the default path?

Reply:
> Is it ever desirable for dune to distinguish between the case where lock is omitted and the case where it's explicitly set to the default path?

There's a couple of situations where the difference is useful:

1. A lock file that hasn't been explicitly specified and is absent is not an error. While a lock file that has been explicitly specified and is absent is an error.
2. When the types reflect exactly what the user has written, you can use them for serialization.

The 2nd point is a bit more general, so I tend to writing things in this style. However, 1. is the important one here.